Alias outputs doc (#1153)
iotaledger/iota-wiki: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx

---
title: Destroy Alias Output
description: 'How to programmatically destroy an alias output'
image: /img/logo/iota_mark_light.png
keywords:
  - how to
  - destroy
  - alias
  - output
  - nodejs
  - python
  - rust
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> You can destroy an alias output by ID. If the alias still owns any outputs when you try to destroy it, you will get an error.

Reviewer: ```suggestion
You can destroy an Alias Output using its ID. You will get an error if the alias still owns any outputs when you try to destroy it.
```

> You can destroy an alias output by ID. If the alias still owns any outputs when you try to destroy it, you will get an error.

Reviewer 2: This is not correct, whether an alias output owns other outputs or not does matter when you are destroying one. There is nothing in place from a protocol level that would prevent you from burning/deleting an alias output of which its address are still present in other addresses. Likewise, you could also theoretically create outputs which are owned by an Alias address which does not even exist.

Can you maybe point out how you came to this conclusion?

## Example Code

<Tabs groupId="language" queryString>
<TabItem value="rust" label="Rust">

The following example will:

1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
   [created in the first guide](../accounts-and-addresses/create-account.mdx).

> 2. Destroy an alias output transaction by calling the [`Account.burn()`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.burn) function.

Reviewer: ```suggestion
2. Destroy an Alias Output transaction by calling the [`Account.burn()`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.burn) function.
```

<div className={'hide-code-block-extras'}>

```rust reference
https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/destroy.rs#L44
```

</div>

> 3. Retry transaction until included [`Transaction`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.retry_transaction_until_included) that was sent.

Reviewer: ```suggestion
3. Retry the [`Transaction`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.retry_transaction_until_included) until included.
```

<div className={'hide-code-block-extras'}>

```rust reference

> https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/destroy.rs#L47

Reviewer: ```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/destroy.rs#L47-L49
```

```

</div>

</TabItem>
<TabItem value="typescript-node" label="Typescript (Node.js)">

The following example will:

1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).

> 2. Destroy an alias output transaction by calling the [`Account.preparedestroyalias()`](../../references/nodejs/classes/Account.md#preparedestroyalias) function.

Reviewer: ```suggestion
2. Prepare the transaction to destroy an Alias Output by calling the [`Account.preparedestroyalias()`](../../references/nodejs/classes/Account.md#preparedestroyalias) function, and then [send](../../references/nodejs/classes/Account.md#send) it.
```

<div className={'hide-code-block-extras'}>

```typescript reference

>     https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/destroy.ts#L56

Reviewer: ```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/destroy.ts#L56-L58
```

```

</div>

> 3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account/#retrytransactionuntilincluded) that was sent.

Reviewer 3: ```suggestion
3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account.md#retrytransactionuntilincluded) that was sent.
```

<div className={'hide-code-block-extras'}>

```typescript reference
https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/destroy.ts#L63
```

</div>

</TabItem>
<TabItem value="python" label="Python">

1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).

> 2. Destroy an alias output transaction by calling the [`Account.prepare_destroy_alias()`](../../references/python/iota_sdk/wallet/account.md#prepare_destroy_alias) function.

Reviewer: ```suggestion
2. Destroy the transaction that will destroy the alias output by calling the [`Account.prepare_destroy_alias()`](../../references/python/iota_sdk/wallet/account.md#prepare_destroy_alias) function, and then [send](../../references/python/iota_sdk/wallet/account.md#send) it.
```
